### DISCUSSION TITLE
Usage of OQS-Provider in openSSL 3.0

### DISCUSSION THREAD

**Discussion by stfries** [open-quantum-safe/oqs-provider#333] [Q&A] [answered]
Hi,

hope it is not a silly question. 
The oqs-provider is available in openSSL 3.2. For some use cases we still utilize openSSL 3.0 and I was wondering if there are problems known when using the oqs-provider together with openSSL 3.0. Right from the top of my head I don't see changes in the provider interface from version 3.0 to 3.2 but the readme mentions that there were some glitches addressed for CMS. We are targeting CMS signing, hence the question. 

Appreciate any hint.

Bets regards
Steffen

**Comment by baentsch** [comment, marked as answer]
> hope it is not a silly question.

No, but a relevant one. The answer is documented, though: https://github.com/open-quantum-safe/oqs-provider#3031

> Right from the top of my head I don't see changes in the provider interface from version 3.0 to 3.2 but the readme mentions that there were some glitches addressed for CMS. We are targeting CMS signing, hence the question.

The APIs didn't change, `oqsprovider` didn't change, but [there were OpenSSL flaws fixed](https://github.com/openssl/openssl/pull/17733) that didn't get backported to 3.0. Hence we also do not test CMS in downlevel OpenSSL versions: https://github.com/open-quantum-safe/oqs-provider/blob/eb80076d1683121db35fabf8c1b0c6f9125c1b6d/scripts/oqsprovider-cmssign.sh#L50-L53

**Reply by stfries** [reply]
Thank you for the swift reply. It essentially means we need to do the back porting and related testing to ensure it works as intended.
